File: insercion.js
function insertionSort(a) {

    for (let i = 1; i < a.length; i++) {

        let temp = a[i];

        let j = i - 1;

        while (j >= 0 && temp < a[j]) {
            a[j + 1] = a[j];
            j = j - 1;
        }

        a[j + 1] = temp;
    }
}

function printArr(a) {

    for (let i = 0; i < a.length; i++) {
        process.stdout.write(a[i] + " ");
    }
}

let a = [70, 15, 2, 51, 60];

console.log("Antes de ordenar los elementos del arreglo:");
printArr(a);

insertionSort(a);

console.log("\nDespues de ordenar los elementos del arreglo:");
printArr(a);

File: insercion.cpp
#include <iostream>
using namespace std;

void insertionSort(int a[], int s)
{
    for (int i = 1; i < s; i++)
    {
        int temp = a[i];

        int j = i - 1;

        while (j >= 0 && temp < a[j])
        {
            a[j + 1] = a[j];
            j = j - 1;
        }

        a[j + 1] = temp;
    }
}

void printArr(int a[], int s)
{
    for (int i = 0; i < s; i++)
    {
        cout << a[i] << " ";
    }
}

int main()
{
    int a[] = {70, 15, 2, 51, 60};

    int s = sizeof(a) / sizeof(a[0]);

    cout << "Antes de ordenar los elementos del arreglo: " << endl;
    printArr(a, s);

    insertionSort(a, s);

    cout << "\nDespues de ordenar los elementos del arreglo: " << endl;
    printArr(a, s);

    return 0;
}